### PR TITLE
[Bug] Fix bug that nondeterministic functions should not be rewrote in create view stmt

### LINKED ISSUE #6095: [Bug] The non-deterministic constant function in create view statement was rewritten
**Describe the bug**
```
create view v1 as select * from tbl1 where hour(now()) > 3;
```

is rewritten to:

```
create view v1 as select * from tbl1 where 23 > 3;
```

The non-deterministic constant function in create view statement should not be rewritten.

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

create view v1 as select  * from tbl where  (hour(now()) > 7); should not be rewrote as:
create view v1 as select  * from tbl where where 11 > 7;

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6095 ) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java [hunk 1]
```diff
@@ -25,14 +25,14 @@
 import org.apache.doris.analysis.LikePredicate;
 import org.apache.doris.builtins.ScalarBuiltins;
 
+import org.apache.logging.log4j.LogManager;
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:28]: Why change the order of import?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -216,6 +216,11 @@
 import org.apache.doris.transaction.PublishVersionDaemon;
 import org.apache.doris.transaction.UpdateDbUsedDataQuotaDaemon;
 
+import org.apache.commons.collections.CollectionUtils;
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java:219]: same as above

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Comment by EmmyMiao87** [comment]
Please rebase master

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateViewStmt.java [hunk 3]
```diff
@@ -63,6 +63,11 @@ public void analyze(Analyzer analyzer) throws UserException {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "CREATE");
         }
 
+        // Do not rewrite nondeterministic functions to constant in create view's def stmt
+        if (ConnectContext.get() != null) {
+            ConnectContext.get().setNotEvalNondeterministicFunction(true);
```
Comment (xy720) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateViewStmt.java:68]: When will it set back to false?

**Pull request merged**